File: credit/src/main/java/ru/patterns/credit/infrastructure/messaging/sender/KafkaLogSender.java
package ru.patterns.credit.infrastructure.messaging.sender;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;
import ru.patterns.credit.infrastructure.messaging.KafkaAsyncSender;
import ru.patterns.credit.shared.kafka.TraceDto;

@Component
public class KafkaLogSender extends KafkaAsyncSender<TraceDto> {
    public KafkaLogSender(KafkaTemplate<String, Object> template,
                          ObjectMapper objectMapper) {
        super(template, objectMapper, "trace-logs-topic");
    }
}

File: core-query/src/main/kotlin/ru/patterns/corequery/utils/log/KafkaLogSender.kt
package ru.patterns.corequery.utils.log

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.stereotype.Component
import reactor.kafka.sender.KafkaSender
import ru.patterns.corequery.utils.KafkaAsyncSender
import ru.patterns.corequery.utils.log.serialization.TraceDto

@Component
class KafkaLogSender(
    kafkaSender: KafkaSender<String, String>,
    objectMapper: ObjectMapper,
) : KafkaAsyncSender<TraceDto>(
    kafkaSender = kafkaSender,
    objectMapper = objectMapper,
    topic = "trace-logs-topic"
)